lib/Doctrine/ORM/Internal/Hydration/AbstractHydrator.php

```diff
@@ -396,94 +410,173 @@ abstract protected function hydrateAllData();
+            // Per definition it cannot be that a field is defined several times and has several values.
+            if (isset($rowData[$fieldName])) {
+                continue;
+            }
+
+            // array_key_exists is more accurate as it accounts for null values
+            // but is slower so we first try a quicker version
+            if (! isset($data[$column]) && ! array_key_exists($column, $data)) {
+                continue;
+            }
+
+            $type = $cacheKeyInfo['type'];
+
+            $value = $type
+                ? $type->convertToPHPValue($data[$column], $this->_platform)
+                : $data[$column];
+
+            if ($value !== null && isset($cacheKeyInfo['enumType'])) {
+                $value = $this->buildEnum($value, $cacheKeyInfo['enumType']);
+            }
+
+            $rowData[$fieldName] = $value;
         }
 
         return $rowData;
     }
 
+    /**
+     * @param mixed[] $data SQL Result Row.
+     *
+     * @psalm-return array<array-key, mixed>
+     */
+    protected function gatherRowDataFromScalarsMapping(array $data)
+    {
+        $scalarsData = [];
+
+        foreach ($this->_rowMappingInfo['scalars'] as $column => $cacheKeyInfo) {
+            $fieldName = $cacheKeyInfo['fieldName'];
+            $type      = $cacheKeyInfo['type'];
+            $value     = $type->convertToPHPValue($data[$column], $this->_platform);
+
+            if ($value !== null && isset($cacheKeyInfo['enumType'])) {
+                $value = $this->buildEnum($value, $cacheKeyInfo['enumType']);
+            }
+
+            $scalarsData[$fieldName] = $value;
+        }
+
+        return $scalarsData;
+    }
+
+    /**
+     * @param mixed[] $data SQL Result Row.
+     *
+     * @psalm-return array<array-key, array{
+     *                       class: mixed,
+     *                       args: array
+     *                   }>
+     *
+     * @psalm-suppress MoreSpecificReturnType Psalm does not detect that args keys is always defined
+     */
+    protected function gatherRowDataFromNewObjectsMapping(array $data)
+    {
+        $newObjectsData = [];
+
+        foreach ($this->_rowMappingInfo['newObjects'] as $column => $cacheKeyInfo) {
+            $argIndex = $cacheKeyInfo['argIndex'];
+            $objIndex = $cacheKeyInfo['objIndex'];
+            $type     = $cacheKeyInfo['type'];
+            $value    = $type->convertToPHPValue($data[$column], $this->_platform);
+
+            if ($value !== null && isset($cacheKeyInfo['enumType'])) {
+                $value = $this->buildEnum($value, $cacheKeyInfo['enumType']);
+            }
+
+            $newObjectsData[$objIndex]['class']           = $cacheKeyInfo['class'];
+            $newObjectsData[$objIndex]['args'][$argIndex] = $value;
+        }
+
+        /** @psalm-suppress LessSpecificReturnStatement Psalm does not detect that args keys is always defined */
```

Comment:
Don't suppress Psalm errors via comments. We usually configure this in psalm.xml.dist.

Reply:
@derrabus I don't manage to suppress it for this specific method inside `psalm.xml`.
And to suppress it for the whole file looks a bit too wide to me.

Do you have any recommendations?
Or should I add this to the baseline?
Or does it mean this is a legitimate use case to you?